### PR TITLE
[CNBC] New url regex

### LINKED ISSUE #14236: CNBC videos not working
## Please follow the guide below

- You will be asked some questions and requested to provide some information, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like

---

### Make sure you are using the *latest* version: run `youtube-dl --version` and ensure your version is *2017.09.15*. If it's not, read [this FAQ entry](https://github.com/rg3/youtube-dl/blob/master/README.md#how-do-i-update-youtube-dl) and update. Issues with outdated version will be rejected.
- [x ] I've **verified** and **I assure** that I'm running youtube-dl **2017.09.15**

### Before submitting an *issue* make sure you have:
- [ x] At least skimmed through the [README](https://github.com/rg3/youtube-dl/blob/master/README.md), **most notably** the [FAQ](https://github.com/rg3/youtube-dl#faq) and [BUGS](https://github.com/rg3/youtube-dl#bugs) sections
- [x ] [Searched](https://github.com/rg3/youtube-dl/search?type=Issues) the bugtracker for similar issues including closed ones

### What is the purpose of your *issue*?
- [ ] Bug report (encountered problems with youtube-dl)
- [x ] Site support request (request for adding support for a new site)
- [ ] Feature request (request for a new functionality)
- [ ] Question
- [ ] Other

---

### The following sections concretize particular purposed issues, you can erase any section (the contents between triple ---) not applicable to your *issue*

---

### If the purpose of this *issue* is a *bug report*, *site support request* or you are not completely sure provide the full verbose output as follows:

Add the `-v` flag to **your command line** you run youtube-dl with (`youtube-dl -v <your command line>`), copy the **whole** output and insert it here. It should look similar to one below (replace it with **your** log inserted between triple ```):

```
C:\youtube-dl>youtube-dl -v https://www.cnbc.com/video/2017/09/15/are-the-homebuilders-showing-cracks.html
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['-v', 'https://www.cnbc.com/video/2017/09/15/are-the-homebuilders-showing-cracks.html']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2017.09.15
[debug] Python version 3.4.4 - Windows-10-10.0.15063
[debug] exe versions: ffmpeg N-87306-g6743351, ffprobe N-87306-g6743351
[debug] Proxy map: {}
[generic] are-the-homebuilders-showing-cracks: Requesting header
WARNING: Falling back on generic information extractor.
[generic] are-the-homebuilders-showing-cracks: Downloading webpage
[generic] are-the-homebuilders-showing-cracks: Extracting information
ERROR: Unsupported URL: https://www.cnbc.com/video/2017/09/15/are-the-homebuilders-showing-cracks.html
Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmpl1klaw4y\build\youtube_dl\YoutubeDL.py", line 776, in extract_info
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmpl1klaw4y\build\youtube_dl\extractor\common.py", line 434, in extract
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\rg3\tmpl1klaw4y\build\youtube_dl\extractor\generic.py", line 2964, in _real_extract
youtube_dl.utils.UnsupportedError: Unsupported URL: https://www.cnbc.com/video/2017/09/15/are-the-homebuilders-showing-cracks.html
```

---

### If the purpose of this *issue* is a *site support request* please provide all kinds of example URLs support for which should be included (replace following example URLs by **yours**):
- Single video: https://www.cnbc.com/video/2017/09/15/are-the-homebuilders-showing-cracks.html
- Single video: https://www.cnbc.com/video/2017/09/15/tesla-just-posted-its-best-week-in-over-a-year.html

Note that **youtube-dl does not support sites dedicated to [copyright infringement](https://github.com/rg3/youtube-dl#can-you-add-support-for-this-anime-video-site-or-site-which-shows-current-movies-for-free)**. In order for site support request to be accepted all provided example URLs should not violate any copyrights.

---

### Description of your *issue*, suggested solution and other information

CNBC videos not downloading. New to this so unsure if I'm doing something wrong or not supported (though site is on supported list). Thanks for the help.

### PR DESCRIPTION
**Pull request by JCGoran**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Fixes #14236, download of CNBC videos should work now.

### CODE REVIEW

File: youtube_dl/extractor/cnbc.py [hunk 1]
```diff
@@ -25,12 +30,23 @@ class CNBCIE(InfoExtractor):
     }
 
     def _real_extract(self, url):
-        video_id = self._match_id(url)
+        mob = re.search(self._VALID_URL, url)
+        video_id = mob.group('title')
```
Comment (dstftw) [inline youtube_dl/extractor/cnbc.py]: `self._match_id`.

File: youtube_dl/extractor/cnbc.py [hunk 2]
```diff
@@ -25,12 +30,23 @@ class CNBCIE(InfoExtractor):
...
+        webpage = self._download_webpage(url, video_id)
+        mobj = re.search(
+                r'mpscall\s*=\s*(?P<json_data>[^)]+),\s*mpsopts',
+            webpage)
```
Comment (dstftw) [inline youtube_dl/extractor/cnbc.py]: `self._search_regex`.

File: youtube_dl/extractor/cnbc.py [hunk 3]
```diff
@@ -25,12 +30,23 @@ class CNBCIE(InfoExtractor):
...
+
+        urls_info = self._parse_json(
+            mobj.group('json_data'), video_id, transform_source = js_to_json)
+
         return {
             '_type': 'url_transparent',
             'ie_key': 'ThePlatform',
             'url': smuggle_url(
-                'http://link.theplatform.com/s/gZWlPC/media/guid/2408950221/%s?mbr=true&manifest=m3u' % video_id,
+                'http://link.theplatform.com/s/gZWlPC/media/guid/2408950221/%s?mbr=true&manifest=m3u' % urls_info.get('content_id'),
```
Comment (dstftw) [inline youtube_dl/extractor/cnbc.py]: `urls_info['content_id']`.

**Comment by JCGoran** [comment]
Requested changes have been implemented, anything else?

**Comment by fenollp** [comment]
CNBC links don't work with version 2018.06.04. I tested this branch with [this link](https://www.cnbc.com/video/2018/06/04/microsoft-ceo-on-github-acquisition.html) and this patch works :)

Fixes: https://github.com/rg3/youtube-dl/issues/16535 https://github.com/rg3/youtube-dl/issues/14236 
**Maybe** fixes but *probably not*: https://github.com/rg3/youtube-dl/issues/13222 https://github.com/rg3/youtube-dl/issues/14193 https://github.com/rg3/youtube-dl/issues/9011 

Note: CI failures seem unrelated.
Thanks @JCGoran & @dstftw !

**Comment by shirishag75** [comment]
any updates to the fixes ?

**Pull request closed without merging**